src/overlays/actors/ovl_Obj_Boyo/z_obj_boyo.c

```diff
@@ -21,20 +41,203 @@ const ActorInit Obj_Boyo_InitVars = {
     (ActorFunc)ObjBoyo_Update,
     (ActorFunc)ObjBoyo_Draw
 };
-*/
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Init.s")
+#endif
+
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Init.s")
+// D_809A6170
+static ColliderCylinderInit sColliderCylinderInit = {
+    {
+        /* ColliderInit base */
+        0xA,  /* u8 coltype */
+        0,    /* u8 atFlags */
+        0xD,  /* u8 acFlags */
+        0x39, /* u8 ocFlags1 */
+        0x20, /* u8 ocFlags2 */
+        1     /* u8 shape */
+    },
+    {
+        /* ColliderInfoInit info */
+        0, /* u8 elemType */
+        {  /* ColliderTouchInit toucher */
+          0 /* u32 dmgFlags */, 0 /* u8 effect */, 0 /* u8 defense */ },
+        {
+            /* ColliderBumpInit  bumper */
+            0x01CBFFBE /* u32 dmgFlags */, 0 /* u8 effect */, 0 /* u8 defense */
+        },
+        0, /* u8 toucherFlags */
+        1, /* u8 bumperFlags */
+        1  /* u8 ocElemFlags */
+    },
+    { /* Cylinder16 dim */
+      0x3C /* s16 radius */,
+      0x8C /* s16 height */,
+      0 /* yShift */,
+      {
+          /* Vec3s pos */
+          0, /* s16 x */
+          0, /* s16 y */
+          0  /*s16 z */
+      } },
+};
+
+// D_809A619C
+static InitChainEntry sInitChainEntry[5] = { 0xB0FC0FA0, 0xB100012C, 0xB104012C, 0x48580064,
+                                             0x00000000 }; /* unable to generate initializer */
+
+typedef void (*ObjBoyo_CollisionHandler)(ObjBoyo* this, Actor* actor);
+ObjBoyo_CollisionHandler D_809A61B0[] = { ObjBoyo_PushPlayer };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Boyo/ObjBoyo_Destroy.s")
+extern s16 D_809A61B4[0xE];
```

Comment:
I think the main issue you are having with update has to do with the fact that these data symbols actually seem to be part of the same one (i.e D_809A61B4 is fake, and is actually just a part of D_809A61B0).
I would recommend in variables.txt to remove the `0x809A61B4:("D_809A61B4","UNK_TYPE2","",0x2),` line and change the `D_809A61B0` line to `0x809A61B0:("D_809A61B0","UNK_TYPE1","",0xC),` (the 0x1 changed to 0xC). If you disasm again, remove the data here for the time being, and run m2c again it should give much better output.

Reply:
Are you still looking at matching `Update`? Would you like some more help?

Reply:
I am really struggling do not really know where to start to make it match